jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/BlockingTest.java

```diff
@@ -262,52 +194,33 @@ public void testSharedNoRun() throws Exception
     @Test
     public void testSharedSucceededBlock() throws Exception
     {
-        long start;
         try (Blocker.Callback callback = _shared.callback())
         {
             callback.succeeded();
-            start = NanoTime.now();
             callback.block();
         }
-        assertThat(NanoTime.millisSince(start), lessThan(500L));
     }
 
     @Test
     public void testSharedBlockSucceeded() throws Exception
     {
-        long start;
-        try (Blocker.Callback callback = _shared.callback())
+        try (Blocker.Callback callback = _shared.callback();
+             AssertingThread thread = new AssertingThread(() ->
+             {
+                 await().atMost(5, TimeUnit.SECONDS).until(main::getState, Matchers.is(Thread.State.WAITING));
+                 callback.succeeded();
+             }))
         {
-            final CountDownLatch latch = new CountDownLatch(1);
-
-            new Thread(() ->
-            {
-                latch.countDown();
-                try
-                {
-                    TimeUnit.MILLISECONDS.sleep(100);
-                }
-                catch (Exception e)
-                {
-                    e.printStackTrace();
-                }
-                callback.succeeded();
-            }).start();
+            thread.start();
 
-            latch.await();
-            start = NanoTime.now();
             callback.block();
         }
-        long elapsed = NanoTime.millisSince(start);
-        assertThat(elapsed, greaterThan(10L));
-        assertThat(elapsed, lessThan(1000L));
     }
 
     @Test
     public void testSharedFailedBlock()
     {
-        final Exception ex = new Exception("FAILED");
-        long start = Long.MIN_VALUE;
+        Exception ex = new Exception("FAILED");
```

Comment:
Another place `assertThrows()` could be used

Reply:
Changed.